rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -600,26 +601,36 @@ private void saveCoinbaseInformations() {
             safeSaveToRepository(getStorageKeyForCoinbaseInformation(blockHash), data, BridgeSerializationUtils::serializeCoinbaseInformation));
     }
 
-    public boolean isFastBridgeFederationDerivationHashUsed(Sha256Hash derivationArgsHash) {
+    public boolean isFastBridgeFederationDerivationHashUsed(Sha256Hash btcTxHash, Sha256Hash derivationArgsHash) {
         if (!activations.isActive(RSKIP176)) {
             return false;
         }
-        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationArguments(derivationArgsHash));
+
+        if (btcTxHash == null || derivationArgsHash == null) {
+            return false;
+        }
+
+        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationArguments(btcTxHash, derivationArgsHash));
 
         return ((data != null) && (data.length == 1) && (data[0] == FAST_BRIDGE_FEDERATION_SCRIPT_HASH_TRUE_VALUE));
     }
 
-    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash derivationArgsHash) {
+    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash btcTxHashToSave, Sha256Hash derivationArgsHash) {
         if (activations.isActive(RSKIP176)) {
+            fastBridgeBtcTxHashToSave = btcTxHashToSave;
             fastBridgeDerivationArgumentsScriptHashToSave = derivationArgsHash;
```

Comment:
Why `fastBridgeDerivationArgumentsScriptHashToSave`? Should not be `fastBridgeDerivationArgumentsHashToSave`?